packages/editor/src/components/post-author/index.js

```diff
@@ -1,71 +1,137 @@
+/**
+ * External dependencies
+ */
+import { debounce } from 'lodash';
+
 /**
  * WordPress dependencies
  */
+import { useState, useMemo } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
-import { withInstanceId, compose } from '@wordpress/compose';
-import { Component } from '@wordpress/element';
-import { withSelect, withDispatch } from '@wordpress/data';
-import { decodeEntities } from '@wordpress/html-entities';
+import { ComboboxControl } from '@wordpress/components';
 
 /**
  * Internal dependencies
  */
 import PostAuthorCheck from './check';
 
-export class PostAuthor extends Component {
-	constructor() {
-		super( ...arguments );
+let latestAvailableAuthors;
+function PostAuthor() {
+	const authors = useSelect( ( select ) => select( 'core' ).getAuthors() );
+	const postAuthor = useSelect( ( select ) =>
+		select( 'core' ).getAuthor(
+			select( 'core/editor' ).getEditedPostAttribute( 'author' )
+		)
+	);
+
+	const dispatch = useDispatch();
+
+	const authorsForField = useMemo( () => {
+		return authors.map( ( author ) => {
+			return {
+				key: author.id,
+				name: author.name,
+				id: author.id,
+			};
+		} );
+	}, [ authors ] );
 
-		this.setAuthorId = this.setAuthorId.bind( this );
+	// Ensure the current author is included in the initial dropdown list.
+	let foundAuthor = authorsForField.findIndex(
+		( author ) => postAuthor?.id === author.id
+	);
+
+	// The currently field value.
+	const [ fieldValue, setFieldValue ] = useState( postAuthor?.name );
```

Comment:
If you are keeping the value in the local state, why isn't the component controlled?

Reply:
The component already controls its own state and I had trouble trying to make it a controlled component when setting a value, the error was about changing from a controlled to non controlled component. passing a value doesn't seem to work as expected with the controlled combobox, maybe something to address in a follow up? 

The main reason to track state here is to trigger a search when fieldValue changes.

Reply:
Have you tried passing `fieldValue` to the Combobox?

Reply:
I'm not sure how this would help? Can you explain what you mean here?